### PR TITLE
Add log download entrypoint

### PR DESCRIPTION
**Pull request by joshedney**
## Goal

Add a way to download the device session logs from BrowserStack to reduce the number of clicks needed. This is "phase 1" of the update. Further updates will be made to enable this to be run from the CLI during the pipeline.

## Design

 Rather than embed this into the normal run. I've moved it to its own executable to allow end users to run the command from the command line at their leisure.

## Changeset

Add executable to download session logs from BrowserStack

## Tests

Tested locally.

### CODE REVIEW

File: bin/download-logs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+require 'json'
+require 'uri'
+require 'fileutils'
+
+class DownloadLogEntry
+  def start(args)
+    @args = args.dup
+    options = args
+    username = ENV['BROWSER_STACK_USERNAME']
+    access_key = ENV['BROWSER_STACK_ACCESS_KEY']
```
Comment (joshedney) [inline bin/download-logs:17]: I've had to do it this way as I couldn't get the `parser`/`processor` to pick them up, even though they are set in the environment

File: bin/download-logs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+require_relative '../lib/maze/browser_stack_utils'
+require_relative '../lib/maze/logger'
+require 'net/http'
+require 'json'
+require 'uri'
+require 'fileutils'
+
+class DownloadLogEntry
+  def start(args)
+    # TODO change this to --file and --build-id
```
Comment (twometresteve) [inline bin/download-logs:14]: For this PR can we print a simple usage message if either no args of `--help` is given?

File: bin/download-logs [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+                                                    access_key,
+                                                    opts[:build_id]
+
+    $logger.info "Getting logs for build: #{opts[:build_id]}"
+
+    build_info.each_with_index do |session, index|
+      $logger.info "Downloading Device Logs for Session: #{index + 1} to #{Dir.pwd}/maze_output"
+      Maze::BrowserStackUtils.download_log username,
+                                           access_key,
+                                           "#{opts[:build_id]}-#{index + 1}",
```
Comment (twometresteve) [inline bin/download-logs:57]: Assigning this to a local variable first (and using on L64 too) would make it more readable overall.

File: lib/maze.rb [hunk 4]
```diff
@@ -7,7 +7,7 @@
 # Glues the various parts of MazeRunner together that need to be accessed globally,
 # providing an alternative to the proliferation of global variables or singletons.
 module Maze
-  VERSION = '6.9.6'
+  VERSION = '6.9.7'
```
Comment (twometresteve) [inline lib/maze.rb:10]: Actually I'd go for `6.10.0`:
- major - breaking break
- minor - new feature (which this is)
- patch - bug fix (what that is exactly is open to debate sometimes)

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,12 @@
 
 ## Fixes
 
+- Add method to download device logs from BrowserStack` [#344](https://github.com/bugsnag/maze-runner/pull/344)
+
+# 6.9.7 - 2022/04/01
+
+## Fixes
+
```
Comment (twometresteve) [inline CHANGELOG.md:10]: This has got a little confused, as 6.9.7 hasn't been released - your PR can go in with it.  I'd rejig to make the whole release as follows (removing the TDB above):
```
# 6.10.0 - 2022/04/01

## Enhancements

- Add method to download device logs from BrowserStack` [#344](https://github.com/bugsnag/maze-runner/pull/344)

## Fixes

- Correct BrowserStack capability to `disableAnimations` [#323](https://github.com/bugsnag/maze-runner/pull/323)

```

**Review by twometresteve — APPROVED** [review]

**Pull request merged**